ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/blacklist-item/blacklist-item.tsx

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/
+
+import { hot } from 'react-hot-loader'
+import * as React from 'react'
+import { keyframes } from '../../styles/styled-components'
+import styled from '../../styles/styled-components'
+import { buttonTypeEnum, Button } from '../button'
+
+type Props = {
+  remove: () => void
+  navigate: () => void
+  itemTitle: string
+  clearing: boolean
+}
+
+const collapseAnimation = (initialHeight: string) => {
+  return keyframes`
+    from {
+      height: ${initialHeight};
+    }
+    to {
+      height: 0px;
+    }
+  `
+}
+
+const Root = styled<Props, 'div'>('div')`
+  display: block;
+  text-align: center;
+  margin-bottom: ${props => props.theme.minimumSpacing};
+  cursor: pointer;
+  height: ${props => props.theme.minimumButtonSize};
+  overflow: hidden;
+  ${props =>
+    props.clearing
+      ? `animation: ${collapseAnimation(props.theme.minimumButtonSize)} 
+      ${props.theme.coreTransitionTime} linear forwards;`
+      : ''};
+`
+
+const ItemDetails = styled<Props, 'div'>('div')`
+  vertical-align: top;
+  padding: 0px ${props => props.theme.minimumSpacing};
+  text-align: center;
+  width: calc(100% - 2 * ${props => props.theme.minimumButtonSize});
+  height: ${props => props.theme.minimumButtonSize};
+  line-height: ${props => props.theme.minimumButtonSize};
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  overflow: hidden;
+  display: inline-block;
+`
+
+const BlacklistItemPresentation = (props: Props) => {
+  return (
+    <Root {...props}>
```

Comment:
I don't think you need to pass props through to styled if you're only using the Theme. Styled's ThemeProvider handles that for you.

Reply:
I need it here, because I need other variables from the props. Changed everywhere else though